pr-checks/checks/MilestoneCheck.ts

```diff
@@ -0,0 +1,57 @@
+import { context } from '@actions/github'
+import { EventPayloads } from '@octokit/webhooks'
+import { Check } from '../Check'
+import { CheckContext, CheckSubscriber } from '../types'
+
+export type MilestoneCheckConfig = {
+	title?: string
+	targetUrl?: string
+	success?: string
+	failure?: string
+}
+
+export class MilestoneCheck extends Check {
+	id = 'milestone'
+
+	constructor(private config: MilestoneCheckConfig) {
+		super()
+	}
+
+	subscribe(s: CheckSubscriber) {
+		s.on(['pull_request', 'pull_request_target'], ['opened', 'synchronize'], async (ctx) => {
+			const pr = context.payload.pull_request as EventPayloads.WebhookPayloadPullRequestPullRequest
+
+			if (pr && pr.milestone) {
+				return this.success(ctx, pr.head.sha)
+			}
+
+			return this.failure(ctx, pr.head.sha)
+		})
+
+		s.on('issues', ['milestoned', 'demilestoned'], async (ctx) => {
+			const issue = context.payload.issue as EventPayloads.WebhookPayloadIssuesIssue
```

Comment:
instead of casting with `as`, you can do:

```
const issue = context.payload.issue
if (!issue || issue.pull_request) {
	return
}

Reply:
Sort of fixed. Kept the `as` for now since I'll probably need it soon in separate PR.